### PR TITLE
Docs: Add linting to rule doc example code blocks (2 of 3) (refs #2271)

### PR DESCRIPTION
**Pull request by IanVS**
The second half of the rules will be added in this PR. I'll be committing periodically and will squash at the end.  

The final PR will be adding `eslint-plugin-markdown` to `makefile.js`.

### CODE REVIEW

**Comment by ilyavolodin** [comment]
LGTM

**Comment by nzakas** [comment]
Looks like the travis failures are for commit message format.

**Comment by IanVS** [comment]
Yes, that is intentional so it isn't merged, as this is a WIP.  Will squash to the correct format when ready for merge.

**Comment by nzakas** [comment]
Oh duh, missed the "WIP" in the title.

**Comment by IanVS** [comment]
It's occurred to me that adding config comments as I've been doing has a possibility of introducing confusion.  Consider this from [`curly`](http://eslint.org/docs/rules/curly) (chosen randomly):

> The following patterns are considered warnings:
> 
> ```js
> /*eslint curly: 2*/
> 
> if (foo) foo++; /*error Expected { after 'if' condition.*/
> 
> ```

This is a common pattern throughout all the docs.  We give examples of "patterns that are considered warnings" and then patterns that are "not considered warnings"

That was all well and good, but now I've added configuration comments which all have the severity set to `2` (i.e. errors).  You can probably see where I'm going with this.  We are using the term "warning" but showing example of code that will "error".

Am I being paranoid here, or does this have the possibility to be confusing, especially to new users (those most likely to read the docs)?  Would it be better to use a word like "errors" or "problems" instead?

**Comment by ilyavolodin** [comment]
I personally don't see it as an issue that would cause confusion. But I do see your point.

**Comment by nzakas** [comment]
I'm agnostic about this. I don't think warnings vs. errors is all that important a distinction for rule documentation, but I'm also not averse to changing to "problems." The important thing to communicate is that we will tell you something is wrong about this code.

**Comment by btmills** [comment]
While I haven't seen anyone confused by it, "problems" is a good way to eliminate any chance of that. If it could be done with a global find & replace "considered warnings" -> "considered problems" in thirty seconds it's probably worth it.

**Comment by IanVS** [comment]
OK I'll change them to "problems".  I think it wasn't confusing before, because we weren't including the config comments.  But now saying something will be a warning when it's explicitly configured to be an error is contradictory.

**Comment by IanVS** [comment]
OK, phew, ready for review. :sweat: 

The third and final PR in this set will be adding linting to the `makefile.js`.

**Comment by ilyavolodin** [comment]
LGTM

File: docs/rules/wrap-iife.md [hunk 1]
```diff
@@ -47,3 +31,37 @@ This configures the rule to enforce wrapping always the function expression.
...
+
+The following patterns adhere to this rule:
+
+```js
+/*eslint wrap-iife: [2, "inside"]*/
+
+var x = (function () { return { y: 1 };})();
+```
+
+```json
```
Comment (btmills) [inline docs/rules/wrap-iife.md]: Is this one meant to be `js` as well?

Comment (IanVS) [inline docs/rules/wrap-iife.md]: Ooo, good catch!

**Comment by IanVS** [comment]
Anything else needed here?  If so, I'd like to take care of it sooner than later, since this is a large commit and will get out-of-date quickly.

**Pull request merged**